tests/api/test_states.py

```diff
@@ -492,3 +499,356 @@ async def test_cancelling_non_running_flow_run_returns_cancelled(self, flow_run_
         result = await api.states.cancel_flow_run(flow_run_id=flow_run_id)
         assert result.flow_run_id == flow_run_id
         assert result.state == "Cancelled"
+
+
+class TestQueueFlowRun:
+    async def test_doeesnt_transition_on_submitted_transition_with_no_slots(
+        self,
+        flow_id: str,
+        labeled_flow_run_id: str,
+        flow_concurrency_limit: models.FlowConcurrencyLimit,
+    ):
+        """
+        Cloud compatability test; If a flow run is attempting to transition to
```

Comment:
This is currently an inaccurate comment, and will change depending on whether we're okay with Server and Cloud slightly deviating in which runs get Queued.